pkg/logrusutil/logrusutil.go

```diff
@@ -33,24 +37,30 @@ func PropagateJSON(logger *logrus.Logger, jsonLine []byte, header string, begin
 	if err != nil {
 		goto fallback
 	}
+	if err := json.Unmarshal(jsonLine, &fields); err == nil {
+		delete(fields, "level")
+		delete(fields, "msg")
+		delete(fields, "time")
```

Comment:
Could you add comments to explain these deletes

Reply:
I added a comment, and also added a call to set the log timestamp to the original time:

```golang
	entry = entry.WithTime(j.Time)
```

Reply:
I've now also renamed the `logger` variable in the `qemu_driver.go` change to `entry` because its type is `logrus.Entry` and not `logrus.Logger`.